src/senaite/lims/browser/spotlight/jsonapi.py

```diff
@@ -20,68 +23,93 @@ def spotlight_search_route(context, request):
         "portal_catalog",
         "bika_setup_catalog",
         "bika_catalog",
-        # "bika_analysis_catalog"
+        "bika_catalog_worksheet_listing"
     ]
 
     search_results = []
     for catalog in catalogs:
         search_results.extend(search(catalog=catalog))
 
-    def get_state(brain):
-        state = getattr(brain, "review_state", "")
-        if not isinstance(state, basestring):
-            return ""
-        return state
-
-    items = []
-    for brain in search_results:
-        icon = api.get_icon(brain)
-        # avoid 404 errors with these guys
-        if "document_icon.gif" in icon:
-            icon = ""
-
-        id = api.get_id(brain)
-        title = api.get_title(brain)
-
-        items.append({
-            "id": id,
-            "title": title,
-            "title_or_id": title or id,
-            "description": api.get_description(brain),
-            "uid": api.get_uid(brain),
-            "path": api.get_path(brain),
-            "url": api.get_url(brain),
-            "state": get_state(brain),
-            "icon": icon,
-        })
+    # extract the data from all the brains
+    items = map(get_brain_info, search_results)
 
     return {
         "count": len(items),
-        "items": items,
+        "items": sorted(items, key=itemgetter("title")),
+    }
+
+
+def get_brain_info(brain):
+    """Extract the brain info
+    """
+    icon = api.get_icon(brain)
+    # avoid 404 errors with these guys
+    if "document_icon.gif" in icon:
+        icon = ""
+
+    id = api.get_id(brain)
+    uid = api.get_uid(brain)
+    url = api.get_url(brain)
+    path = api.get_path(brain)
+    title = api.get_title(brain)
+    description = api.get_description(brain)
+    parent = api.get_parent(brain)
+    parent_title = api.get_title(parent)
+    parent_url = api.get_url(parent)
+    state = api.get_review_status(brain)
+
+    return {
+        "id": id,
+        "title": title,
+        "title_or_id": title or id,
+        "description": description,
+        "uid": uid,
+        "path": path,
+        "url": url,
+        "parent_title": parent_title,
+        "parent_url": parent_url,
+        "state": state,
+        "icon": icon,
     }
 
 
 def search(query=None, catalog=None):
     """Search
     """
     if query is None:
-        query = make_query()
+        query = make_query(catalog)
     if query is None:
         return []
     return api.search(query, catalog=catalog)
 
 
-def make_query():
+@forever.memoize
+def get_search_index_for(catalog):
+    """Returns the search index to query
+    """
+    searchable_text_index = "SearchableText"
+    listing_searchable_text_index = "listing_searchable_text"
+
+    if catalog == CATALOG_ANALYSIS_REQUEST_LISTING:
+        tool = api.get_tool(catalog)
+        indexes = tool.indexes()
+        if listing_searchable_text_index in indexes:
+            return listing_searchable_text_index
```

Comment:
I like you prioritize `listing_searchable_text` index over `SearchableText`. This is gold bonus!